synapse/rest/admin/__init__.py

```diff
@@ -109,7 +109,11 @@ class VersionServlet(RestServlet):
     def __init__(self, hs: "HomeServer"):
         self.res = {
             "server_version": SYNAPSE_VERSION,
-            "python_version": platform.python_version(),
+            "python_version": (
+                platform.python_version()
+                if not hs.config.server.hide_python_version
+                else "UNKNOWN"
+            ),
```

Comment:
Looking at the PATTERNS above (107) this seems to be part of the admin API. The admin API should not be publicly exposed; it should only be made available in a trusted environment.

Therefore I don't see what you stand to gain by omitting the Python version here. What is your motivation?

Reply:
> Exposure of information about the technology stack should be minimized in production environments to reduce the attack surface.

I suspect it would be as simple to block this in your reverse proxy if you do not want to expose the information.

I wonder if it makes sense to expose this *at all* though honestly. Why do we need an admin API for this?

This was added in #4772, that states:

> This is useful when a Synapse instance is behind a proxy that does not forward the 'Server' header (which also contains Synapse version information).

The `Server` header has never included the Python version though, just the Synapse version.

Additionally we have the `/_matrix/federation/v1/version` endpoint which returns similar info, also not including the Python version:

```json
{"server":{"name":"Synapse","version":"1.93.0"}}
```

I'm tempted to say we should just never return the Python version instead of adding more config options.

Reply:
Actually #4772 made it require an admin API, while #5122 removed that restriction.

Should we just nuke it altogether? 💥

Reply:
I like nuking things. Might be worth having a quick read of https://github.com/matrix-org/synapse/issues/3220 though---this seems to be the original motivation.

Reply:
I suspect @richvdh was looking for a way to get a server version for people while providing support? We also recommend not opening that up globally though, so it seems confusing.

Reply:
In #3220, I was looking for a reliable way for people to establish the actual running version of their own Synapse installation, so that they could file useful bug reports (or provide the information when requesting support). Putting it on an admin api is quite a good way to do that, because - as with the rest of the admin api - the recommendation is that it *not* be exposed publicly.

The fact that matrix.org flies in the face of the recommendation by exposing this, along with the whole of the rest of the admin api, is a separate problem. IMHO, if the goal is to reduce publicly visible information about the stack, the correct solution here is to fix that on matrix.org (by blocking the whole of the admin api, or this particular endpoint).

If you do decide to nuke the endpoint, remember to update https://github.com/matrix-org/synapse/blob/develop/docs/upgrade.md#upgrading-synapse and https://github.com/matrix-org/synapse/blob/develop/.github/ISSUE_TEMPLATE/BUG_REPORT.yml, both of which refer to it.

Reply:
(having said all that: I don't believe the Python version, as opposed to the Synapse version, is particularly useful information, and I wouldn't object to its removal.)